snow/engine/common/test_engine.go

```diff
@@ -237,8 +239,11 @@ func (e *EngineTest) Halt(ctx context.Context) {
 		e.HaltF(ctx)
 		return
 	}
-	if e.CantHalt && e.T != nil {
-		e.T.Fatalf("Unexpectedly called Halt")
+	if !e.CantHalt {
+		return
+	}
+	if e.T != nil {
+		require.FailNow(e.T, "Unexpectedly called Halt")
```

Comment:
I believe this is not equivalent to the original test

Reply:
How so?

Reply:
I am wrong, this is good